runtime/lua/vim/filetype.lua

```diff
@@ -49,8 +50,25 @@ local extension = {
   art = "art",
   asciidoc = "asciidoc",
   adoc = "asciidoc",
+  asa = function(path, bufnr)
+    if vim.g.filetype_asa then
+      return vim.g.filetype_asa
+    else
+      return "aspvbs"
+    end
+  end,
   ["asn1"] = "asn",
   asn = "asn",
+  asp = function(path, bufnr)
+    if vim.g.filetype_asp then
+      return vim.g.filetype_asp
+    elseif getline(bufnr, 1):find("perlscript")
+      or getline(bufnr, 2):find("perlscript")
+      or getline(bufnr, 3):find("perlscript") then
```

Comment:
I think we should modify `getline` to take an optional third argument `end_lnum` for cases like this. That way we can instead use

```lua
elseif getline(bufnr, 1, 3):find("perlscript") then
```

Reply:
That's indeed much nicer, especially for monstrosities like https://github.com/vim/vim/blob/1fc6ea9bf3548b578676331f5eac1f7e0611c268/runtime/filetype.vim#L526

Reply:
this is really much better